src/app/Http/Controllers/API/PresentationUserController.php

```diff
@@ -25,7 +25,26 @@ class PresentationUserController extends Controller
      */
     public function subscribe(Presentation $presentation, User $user)
     {
-        //
+        // check that the user isn't already subscribed to the presentation
+        if ($presentation->users()->find($user)) {
+            return response()->json([
+                'success' => false,
+                'message' => "Already subscribed",
+            ], 409);
+        }
```

Comment:
normaly you could avoid that because of the db constraints, so in theory when you do the attach and the tupple is already present into the presentation_user table laravel should return an error with a message.
If that doesn't work you certainly could do it with a validator that will check if the tupple is unique or something like that and return the apropriate error

Reply:
I tested it and it doesn't generate an error. It adds a new relationship